pepper-apis/src/main/java/org/broadinstitute/ddp/service/actvityinstancebuilder/form/FormInstanceCreator.java

```diff
@@ -92,12 +94,35 @@ public void updateBlockStatuses(AIBuilderContext ctx) {
                 ctx.getParams().getInstanceSummary());
     }
 
-    public void renderContent(AIBuilderContext ctx, Renderable.Provider<String> rendered) {
+    /**
+     * The rendering done in 2 runs - in order to correctly calculate templates depending on results of other templates
+     * (for example with using of Velocity method $ddp.answer())
+     */
+    public void renderContent(AIBuilderContext ctx) {
+        Map<Long, String> renderedTemplates = new TreeMap<>();
+
+        // 1st run
+        renderTemplates(ctx, renderedTemplates);
+        applyRenderedTemplates(ctx, renderedTemplates);
```

Comment:
Just so I understand: Is this first call to "apply" needed because we need to associate the option label text, so that on the second run we'll see it in the `$ddp.answer()` substitutions?

Reply:
Yes, for this purpose.

Reply:
In the design doc there is a requirement: "Another limitation is that we can substitute answers into things like activity name and title, but not contents/questions.".
I think that 2 runs of templates applying should provide help to solve this problem.

Reply:
Right. That problem is captured in another ticket, so I wasn't expecting it here. But that's fine.

Reply:
Now that I think about it, maybe we can avoid this need for 2 runs if we tried passing in the "form response" object and use the "render from form response" approach in `RenderValueProvider` instead? I suspect we're having this issue because we're looking for option label in the "form instance", which is not fully created yet because that's the thing we're trying to build in the "activity instance builder"! With "form response", we shouldn't have this problem because we're pulling the label right from the template and activity definition.

Reply:
Fixed: removed 2 runs and replaced "form instance" with "form response".